test/e2e/e2e_test.go

```diff
@@ -163,105 +172,120 @@ func TestChannelTracing(t *testing.T) {
 func TestSmokePullSubscription(t *testing.T) {
 	cancel := logstream.Start(t)
 	defer cancel()
-	SmokePullSubscriptionTestImpl(t)
+	SmokePullSubscriptionTestImpl(t, workloadIdentity, pubsubServiceAccount)
 }
 
 // TestPullSubscriptionWithTarget tests we can knock down a target.
 func TestPullSubscriptionWithTarget(t *testing.T) {
 	cancel := logstream.Start(t)
 	defer cancel()
-	PullSubscriptionWithTargetTestImpl(t)
+	PullSubscriptionWithTargetTestImpl(t, workloadIdentity, pubsubServiceAccount)
 }
 
 // TestSmokeCloudPubSubSource makes sure we can run tests on the CloudPubSubSource.
 func TestSmokeCloudPubSubSource(t *testing.T) {
 	cancel := logstream.Start(t)
 	defer cancel()
-	SmokeCloudPubSubSourceTestImpl(t)
+	SmokeCloudPubSubSourceTestImpl(t, workloadIdentity, pubsubServiceAccount)
 }
 
 // TestCloudPubSubSourceWithTarget tests we can knock down a target from a CloudPubSubSource.
 func TestCloudPubSubSourceWithTarget(t *testing.T) {
 	cancel := logstream.Start(t)
 	defer cancel()
-	CloudPubSubSourceWithTargetTestImpl(t, false /*assertMetrics */)
+	CloudPubSubSourceWithTargetTestImpl(t, false /*assertMetrics */, workloadIdentity, pubsubServiceAccount)
 }
 
 // TestCloudPubSubSourceStackDriverMetrics tests we can knock down a target from a CloudPubSubSource and that we send metrics to StackDriver.
 func TestCloudPubSubSourceStackDriverMetrics(t *testing.T) {
 	t.Skip("See issues https://github.com/google/knative-gcp/issues/317 and https://github.com/cloudevents/sdk-go/pull/234")
 	cancel := logstream.Start(t)
 	defer cancel()
-	CloudPubSubSourceWithTargetTestImpl(t, true /*assertMetrics */)
+	CloudPubSubSourceWithTargetTestImpl(t, true /*assertMetrics */, workloadIdentity, pubsubServiceAccount)
 }
 
 // TestBrokerWithPubSubChannel tests we can knock a Knative Service from a broker with PubSub Channel.
 func TestBrokerWithPubSubChannel(t *testing.T) {
+	if workloadIdentity {
```

Comment:
Broker test will create the Broker and use `client.Core.CreateBrokerOrFail(brokerName, eventingtestresources.WithChannelTemplateForBroker(lib.ChannelTypeMeta))` to create channel, `WithChannelTemplateForBroker` only accept `ChannelTypeMeta`. https://github.com/knative/eventing/blob/93707e9f7466a729d16e5ad3a20fd2c48490c54b/test/lib/resources/eventing.go#L181-L188

Looks like we need to change eventing's `WithChannelTemplateForBroker` method to include channel's `Spec.serviceAccount` in channeltemplate?

Reply:
The Broker v1beta1 spec seems to have deprecated the `ChannelTemplate`, and is using `Config` instead. But the same suggestion as the above, we should pass `Spec` instead of `Spec.serviceAccount`, the code to set it should be at https://github.com/knative/eventing/blob/master/test/lib/creation.go#L176 (I don't know when it's added but it's pretty ugly...)